src/main/java/frc/robot/scoreStates/AmpState.java

```diff
@@ -8,49 +8,88 @@
+        return Commands.runOnce(
                 () -> {
-                    Translation2d ampPose;
+                    isAmpingForward = true;
                     if (isRed()) {
                         ampPose = ScoreStateConstants.AMP_POSE_RED;
                     } else {
                         ampPose = ScoreStateConstants.AMP_POSE_BLUE;
                     }
-                    Rotation2d ampRotation = ScoreStateConstants.AMP_ROTATION_NORMAL;
+                    ampRotation = ScoreStateConstants.AMP_ROTATION_NORMAL;
                     Pose2d botPose = SwerveDrive.getInstance().getBotPose();
-                    double robotRotation = botPose.getRotation().getRadians();
-                    boolean isGripperReversed =
-                            false; // TODO: replace with actual gripper.isForward
-                    double distance = Utils.getDistanceFromPoint(ampPose, botPose);
-                    boolean hasTimeToTurnGripper =
+                    robotRotation = botPose.getRotation().getRadians();
+                    distance = Utils.getDistanceFromPoint(ampPose, botPose);
+                    hasTimeToTurnGripper =
                             distance
                                     > ScoreStateConstants.MIN_DISTANCE_TO_TURN_GRIPPER.in(
                                             Units.Meters);
-
-                    if ((isGripperReversed && robotRotation < 0) || hasTimeToTurnGripper) {
+                    if ((gripper.isReversed() && robotRotation < 0) || hasTimeToTurnGripper) {
+                        isAmpingForward = false;
                         ampRotation = ScoreStateConstants.AMP_ROTATION_REVERSE;
                     }
-                    return AutoBuilder.pathfindToPose(
-                            new Pose2d(ampPose, ampRotation), Constants.AUTO_CONSTRAINTS);
-                },
+                });
+    }
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () ->
+                        AutoBuilder.pathfindToPose(
+                                new Pose2d(ampPose, ampRotation), Constants.AUTO_CONSTRAINTS),
                 Set.of(SwerveDrive.getInstance()));
     }
 
     @Override
-    public Command stateInitialize() {
-        return null;
+    public Command initializeSubsystem() {
+        return Commands.defer(
+                () ->
+                        Commands.parallel(
+                                elevator.setHeight(ElevatorConstants.MAX_HEIGHT),
+                                gripper.setWristPosition(
+                                        isAmpingForward
+                                                ? GripperConstants.WRIST_FORWARD_AMP_POSE
+                                                        .mutableCopy()
+                                                : GripperConstants.WRIST_BACKWARDS_AMP
+                                                        .mutableCopy())),
```

Comment:
These would probably be identical. It's better to save one of them and invert when needed. Another advantage of that is you will need to only adjust one value, not two.
This is relevant to lines 89-91, too.
Also, don't create copies here every run.

Reply:
This constants are not opposite because the gripper is not cable of doing a full 180

Reply:
1. These values could be different, if we see they're identical we'll change it.
2. @rakrakon changed creating copies in f7fab12c0b54b59f7b9db926a23cf9c553a2c6b6.